WooCommerce/UITestsFoundation/Screens/Products/ProductsScreen.swift

```diff
@@ -51,4 +47,10 @@ public class ProductsScreen: ScreenObject {
         XCUIApplication().tables.cells.element(boundBy: index).tap()
         return try SingleProductScreen()
     }
+
+    public func verifyProductNameOnProductsScreen(name: String) {
+        let predicate = NSPredicate(format: "label CONTAINS[c] %@", name)
+        let elementQuery = app.staticTexts.containing(predicate)
+        XCTAssertTrue(elementQuery.count == 1, "Product name does not exist!")
+    }
```

Comment:
There are two attributes we can check more or less easily in products list: Product name and stock status. Both of these attributes are already read from mocks.

This is a hierarchy for a separate product "card" in products list screen (example for two products):

<img width="664" alt="Screenshot 2021-12-14 at 23 53 58" src="https://user-images.githubusercontent.com/73365754/146085445-0aeedd61-0e14-46f9-8c4f-5df3580e9491.png">

I think it's possible to add a check that makes sure that a parent of the StaticText with `name` label contains a StaticText with `stock_status` label.

If done, this will actually cover all attributes we can test. Of course, in this case, it will make sense to rename a method to something like `verifyProductCardShownOnProductsScreen`, and make the method argument a structure that contains both product name a stock status (in similarity to `ProductData` from Android counterpart).

Reply:
One way this can be achieved is creating a sort of dictionary. Like [done](https://github.com/woocommerce/woocommerce-android/blob/d88e609ecc7e6896ddc8eccc1eb3abdc48cb81e6/WooCommerce/src/androidTest/kotlin/com/woocommerce/android/screenshots/util/DataClasses.kt#L18-L22) for Android. When you read products values from mocks, you will need to replace the value of `stock_status` with the one from dictionary.